src/scripts/librarymenu.js

```diff
@@ -155,297 +302,671 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
+        currentDrawerType = "library";
+        (user ? Promise.resolve(user) : connectionManager.user(getCurrentApiClient())).then(function (user) {
+            refreshLibraryInfoInDrawer(user);
+            updateLibraryMenu(user.localUser);
+        });
     }
 
     function getNavDrawerOptions() {
         var drawerWidth = screen.availWidth - 50;
-        return drawerWidth = Math.max(drawerWidth, 240), drawerWidth = Math.min(drawerWidth, 320), {
+        drawerWidth = Math.max(drawerWidth, 240);
+        drawerWidth = Math.min(drawerWidth, 320);
+        return {
             target: navDrawerElement,
             onChange: onMainDrawerSelect,
             width: drawerWidth
-        }
+        };
     }
 
     function loadNavDrawer() {
-        return navDrawerInstance ? Promise.resolve(navDrawerInstance) : (navDrawerElement = document.querySelector(".mainDrawer"), navDrawerScrollContainer = navDrawerElement.querySelector(".scrollContainer"), navDrawerScrollContainer.addEventListener("click", onMainDrawerClick), new Promise(function(resolve, reject) {
-            require(["navdrawer"], function(navdrawer) {
-                navDrawerInstance = new navdrawer(getNavDrawerOptions()), navDrawerElement.classList.remove("hide"), resolve(navDrawerInstance)
-            })
-        }))
-    }
-    var navDrawerElement, navDrawerScrollContainer, navDrawerInstance, mainDrawerButton, headerHomeButton, currentDrawerType, pageTitleElement, headerBackButton, headerUserButton, currentUser, headerSettingsButton, headerCastButton, headerSearchButton, enableLibraryNavDrawer = !layoutManager.tv,
-        skinHeader = document.querySelector(".skinHeader"),
-        requiresUserRefresh = !0,
-        lastOpenTime = (new Date).getTime();
+        if (navDrawerInstance) {
+            return Promise.resolve(navDrawerInstance);
+        }
+
+        navDrawerElement = document.querySelector(".mainDrawer");
+        navDrawerScrollContainer = navDrawerElement.querySelector(".scrollContainer");
+        navDrawerScrollContainer.addEventListener("click", onMainDrawerClick);
+        return new Promise(function (resolve, reject) {
+            require(["navdrawer"], function (navdrawer) {
+                navDrawerInstance = new navdrawer(getNavDrawerOptions());
+                navDrawerElement.classList.remove("hide");
+                resolve(navDrawerInstance);
+            });
+        });
+    }
+
+    var navDrawerElement;
+    var navDrawerScrollContainer;
+    var navDrawerInstance;
+    var mainDrawerButton;
+    var headerHomeButton;
+    var currentDrawerType;
+    var pageTitleElement;
+    var headerBackButton;
+    var headerUserButton;
+    var currentUser;
+    var headerSettingsButton;
+    var headerCastButton;
+    var headerSearchButton;
+    var enableLibraryNavDrawer = !layoutManager.tv;
+    var skinHeader = document.querySelector(".skinHeader");
+    var requiresUserRefresh = true;
+    var lastOpenTime = new Date().getTime();
```

Comment:
Jesus, what's with all these global variables...

Reply:
Not entirely sure. Touching most anything in this file completely breaks the UI though, so I'm not keen on the idea.